core/src/main/scala/caliban/GraphQLResponse.scala

```diff
@@ -34,13 +34,19 @@ private object GraphQLResponseCirce {
 
   private def handleError(err: Any): Json =
     err match {
-      case ExecutionError(_, path, _) if path.nonEmpty =>
+      case ExecutionError(_, path, location, _) if path.nonEmpty =>
+        val locationJson =
+          location.fold(Json.obj())(
+            l => Json.obj("location" -> Json.obj("column" -> l.column.asJson, "line" -> l.line.asJson))
```

Comment:
Ah, one more thing. As mentioned in #156, the spec suggests a format which is slightly different from what you did: `"locations": [ { "line": 6, "column": 7 } ]`.

Reply:
note that `location` is evaluated only when `path` is set. This may not be what you want.

Reply:
Ah you’re right, though ive never actually seen multiple locations in the wild. I guess I should change it to be a `List` instead of an `Option` too

Reply:
I mean `path` and `location` are presumably independent of each other therefore the guard `if path.nonEmpty =>` looks to be dangerous as it impacts evaluation of `location`. One could consider `location` being non-empty and `path` being empty even though it's more of an intellectual exercise than a "reality".